Traducir.Web/src/App/Components/SuggestionsHistory.tsx

```diff
@@ -42,69 +44,27 @@ export default class SuggestionsHistory extends React.Component<ISuggestionsHist
-                <span className="font-weight-bold">Suggestion:</span> <pre className="d-inline">{sug.suggestion}</pre>
-            </div>
-            <div>
-                <span className="font-weight-bold">State:</span> <span className={`badge ${this.getBadgeClassFromState(sug.state)}`}>{suggestionStateToString(sug.state)}</span>
-            </div>
-            <table className="table">
-                <thead>
-                    <tr>
-                        <th>Event</th>
-                        <th>User</th>
-                        <th>Comment</th>
-                        <th>Date</th>
-                    </tr>
-                </thead>
-                <tbody>
-                    {sug.histories.map(sugHistory => <tr key={sugHistory.id}>
-                        <td>
-                            {suggestionHistoryTypeToString(sugHistory.historyType)}
-                        </td>
-                        <td>
-                            <a
-                                href={`https://${this.props.config.siteDomain}/users/${sugHistory.userId}`}
-                                target="_blank"
-                                title={`at ${sug.creationDate} UTC`}
-                            >
-                                {sugHistory.userName}
-                            </a>
-                        </td>
-                        <td>
-                            {sugHistory.comment}
-                        </td>
-                        <td>
-                            {sugHistory.creationDate}
-                        </td>
-                    </tr>)}
-                </tbody>
-            </table>
-        </div>);
-    }
-
-    public getBadgeClassFromState(state: StringSuggestionState): string | undefined {
-        switch (state) {
-            case StringSuggestionState.Created:
-                return "badge-secondary";
-            case StringSuggestionState.ApprovedByReviewer:
-            case StringSuggestionState.ApprovedByTrustedUser:
-                return "badge-success";
-            case StringSuggestionState.Rejected:
-                return "badge-danger";
-            case StringSuggestionState.DeletedByOwner:
-                return "badge-dark";
-        }
+        return <div>
+            <SuggestionHistoryFilters
+                userid={this.props.currentUser.id}
+            />
+            <SuggestionHistoryTable
+                suggestions={this.state.suggestions}
+                config={this.props.config}
+            />
+        </div>;
     }
 
     public async userChanged(location: Location): Promise<void> {
         const userId = location.pathname.split("/").pop();
+        const filterId = location.search.substring(location.search.lastIndexOf("=") + 1);
         try {
-            const r = await axios.get<ISOStringSuggestion[]>(`/app/api/suggestions-by-user/${userId}`);
+            const r = await axios.get<ISOStringSuggestion[]>(`/app/api/suggestions-by-user/${userId}`,
```

Comment:
Filters are post. this is get, get needs parameteres. You mean the code all in 1 line? this is related to the first comment you left, query string should be do that way (unless Is there other way I didn't find out).  Axios get receive a parameter objet.

Reply:
I was just saying you should do

```
const r = await axios.get<ISOStringSuggestion[]>(`/app/api/suggestions-by-user/${userId}`, { params: { filterId } })
```

instead of splitting the params on several lines

Reply:
:)